providers/google.go

```diff
@@ -222,3 +265,22 @@ func (g *GoogleOp) VerifyProvider(ctx context.Context, idt []byte, cic *clientin
 func (g *GoogleOp) HookHTTPSession(h http.HandlerFunc) {
 	g.httpSessionHook = h
 }
+
+// FindAvaliablePort attempts to open a listener on localhost until it finds one or runs out of redirectURIs to try
+func FindAvaliablePort(redirectURIs []string) (*url.URL, net.Listener, error) {
+	var ln net.Listener
+	var lnErr error
+	for _, v := range redirectURIs {
+		redirectURI, err := url.Parse(v)
+		if err != nil {
+			return nil, nil, fmt.Errorf("malformed redirectURI specified, redirectURI was %s", v)
+		}
+
+		lnStr := fmt.Sprintf("localhost:%s", redirectURI.Port())
+		ln, lnErr = net.Listen("tcp", lnStr)
```

Comment:
We use `127.0.0.1` explicitly as the host instead of `localhost` in the listener string in `opk-ssh`, not sure if that makes any difference though given that `localhost` should map to that

Reply:
I'm using localhost here because localhost is used in the redirectURI configured in Google. I don't think it should make any difference as this repo has always used localhost in this way.